javascript/src/openai/responses.ts

```diff
@@ -0,0 +1,418 @@
+                },
+              ],
+              type: "message",
+            });
+          }
+          break;
+        }
+      }
+    }
+
+    return responseMessages;
+  }
+
+  /**
+   * Map OpenAI response events to content deltas
+   */
+  private mapEventToContentDeltas(
+    event: OpenAI.Responses.ResponseStreamEvent,
+  ): ContentDelta[] {
+    const contentDeltas: ContentDelta[] = [];
+
+    switch (event.type) {
+      case "response.reasoning_summary_text.delta": {
+        const part: ReasoningPartDelta = {
+          type: "reasoning",
+          reasoning: event.delta,
+          summary: true,
+        };
+        contentDeltas.push({
+          index: event.output_index || 0,
+          part,
+        });
+        break;
+      }
+      case "response.reasoning.delta": {
+        const part: ReasoningPartDelta = {
+          type: "reasoning",
+          reasoning: event.delta as string,
+          summary: false,
+        };
+        contentDeltas.push({
+          index: event.output_index || 0,
+          part,
+        });
+        break;
+      }
+      case "response.reasoning.done": {
+        // When reasoning is done, we get the complete text
+        // The signature should be available in the final response
+        const part: ReasoningPartDelta = {
+          type: "reasoning",
+          reasoning: event.text,
+          summary: false,
+        };
+        contentDeltas.push({
+          index: event.output_index || 0,
+          part,
+        });
+        break;
+      }
+      case "response.output_text.delta": {
+        contentDeltas.push({
+          index: event.output_index || 0,
+          part: {
+            type: "text",
+            text: event.delta,
+          },
+        });
+        break;
+      }
+      // Handle other event types as needed
+    }
+
+    return contentDeltas;
+  }
+
+  /**
+   * Map OpenAI response to SDK ModelResponse
+   */
+  private mapResponseToModelResponse(
+    response: OpenAI.Responses.Response,
+  ): ModelResponse {
+    const content: ModelResponse["content"] = [];
+
+    // Handle output content
+    for (const item of response.output) {
+      // Handle reasoning items
+      if (item.type === "reasoning") {
+        const reasoningPart: ReasoningPart = {
+          type: "reasoning",
+          reasoning: item.summary.map((s) => s.text).join("\n"),
+          summary: true, // Items in output are always summary
+          ...(item.encrypted_content && { signature: item.encrypted_content }),
+        };
+        content.push(reasoningPart);
+      }
+      // Handle message items
+      else if (item.type === "message") {
+        const contentParts = item.content;
+        for (const part of contentParts) {
+          if (part.type === "output_text") {
+            content.push({
+              type: "text",
+              text: part.text,
+            });
+          }
+          // Handle other content types as needed
```

Comment:
we could handle other parts here: image, audio.

Most importantly: tool calling.